pkg/config/expand.go

```diff
@@ -593,7 +593,14 @@ func expandSimpleVariable(
 func expandVariable(
 	context varContext,
 	modToGrp map[string]int) (string, error) {
-	return "", fmt.Errorf("%s: expandVariable", errorMessages["notImplemented"])
+	re := regexp.MustCompile(`\$\((.*?)\)`)
```

Comment:
Shouldn't we be using `anyVariableExp` here?

Reply:
`anyVariableExp` in this context would return the precedent character:

`kernel-devel-$(uname -r)` would return `-$(uname -r)`

That's because anyVariable needs to know what comes first in order to escape when the precedent character is the backslash.

Reply:
I'll state my concerns more explicitly:

1.`anyVariableExp` is the existing regular expression to search for variables.
2. If you provide the string "$(vars.example1) is like \$(vars.example2)", your message will advise the user to escape both `$(vars.example1)` AND `\$(vars.example2)`.

Please use `anyVariableExp` such that we don't encounter (2) and don't provide multiple definitions for the same thing (1). I'm not convinced it's actually true that the matching substring will include the first character, but it's easy to slice out if it does.